src/tespy/networks/networks.py

```diff
@@ -2606,6 +2734,78 @@ def print_components(c, *args):
         else:
             return np.nan
 
+    def print_exergy_conns(self):
+        r"""Print the calculation results of the (specific) physical exergy of
+        the connections to prompt.
+        """
+        df = pd.DataFrame(columns=['e_PH / (kJ / kg)', 'E_PH / MW'])
+        for c in self.conns.index:
+            row = (c.source.label + ':' + c.source_id + ' -> ' +
+                   c.target.label + ':' + c.target_id)
+            row_data = [c.ex_physical/10**3, c.Ex_Physical/10**6]
+            df.loc[row] = row_data
+
+        self.df_exergy_conns = df
+
+        print('\n##### RESULTS (connections) Specific physical exergy and ' +
+              'physical exergy #####')
+        print(tabulate(df, headers='keys', tablefmt='psql', floatfmt='.4f'))
+
+    def print_exergy_comps(self, E_D_min=1000, sort_desc=True):
+        r"""Print the calculations results of the exergy analysis of
+        components and network to prompt.
+
+        - The results are sorted beginning
+        with the component having the biggest exergy destruction by default.
+        - Components with an exergy destruction smaller than 1000 W is not
+        printed to prompt by default
+
+        Parameters
+        ----------
+        E_D_min : float
+            Minimum exergy destruction to be printed to prompt.
+        sort_des : boolean
+            Sort the component results descending by exergy destruction.
+        """
+        df = pd.DataFrame(columns=['label', 'E_P / MW', 'E_F / MW',
+                                   'E_D / MW', 'epsilon', 'y_D,k', 'y*_D,k'])
+        for cp in self.comps.index:
+            if (isinstance(cp, cycle_closer) or isinstance(cp, sink) or
+                    isinstance(cp, source) or isinstance(cp, splitter)):
+                continue
+            if cp.epsilon == -111:
+                continue
+            if not isinstance(cp.E_P, str):
+                cp.E_P = round(cp.E_P / 10**6, 4)
+            if not isinstance(cp.epsilon, str):
+                cp.epsilon = round(cp.epsilon, 4)
+            if cp.E_D < E_D_min:
+                continue
+
+            row_data = [cp.label, cp.E_P, cp.E_F/10**6,
+                        cp.E_D/10**6, cp.epsilon, cp.y_Dk,
+                        cp.ystar_Dk]
```

Comment:
I will move parts of this into the exergy_analysis, so we do not need to loop through the components so many times.